### PR TITLE
Generate SCRF in notebook and read in workflow

### PR DESCRIPTION
**Pull request by tcchiao**

### CODE REVIEW

File: cmip6_downscaling/methods/gard.py [hunk 1]
```diff
@@ -88,134 +91,86 @@ def gard_fit_and_predict(
...
-    bin_center, gamma = gs.vario_estimate(pos=t, field=fields, mesh_type='structured')
-    temporal = gs.Gaussian(dim=1)
-    temporal.fit_variogram(bin_center, gamma, sill=np.mean(np.var(fields, axis=1)))
-
-    return {'temporal': temporal.len_scale, 'spatial': spatial.len_scale}
-
-
-def generate_scrf(
-    data: xr.Dataset,
+def read_scrf(
```
Comment (orianac) [inline cmip6_downscaling/methods/gard.py:94]: i assume by shifting the `calc_correlation_length_scale` and `write_scrf` bits to the notebook the code is identical? like no tweaks along the way that we'd want to keep track of?

File: notebooks/scrf.ipynb [hunk 2]
```diff
@@ -0,0 +1,828 @@
...
+    "        train_period_start=1980,\n",
+    "        train_period_end=1980,\n",
+    "        variables=v,\n",
+    "        chunking_approach=None,\n",
+    "    )[v].isel(time=0)\n",
+    "    template[\"lon\"] = convert_long3_to_long1(template.lon)\n",
+    "    template = template.reindex(lon=sorted(template.lon.values))\n",
+    "    template = template.rename({\"lon\": \"x\", \"lat\": \"y\"})\n",
+    "\n",
+    "    mill_template = xr.DataArray(\n",
```
Comment (orianac) [inline notebooks/scrf.ipynb:493]: a small comment here about this being the point where the miller projection comes into play

File: notebooks/scrf.ipynb [hunk 3]
```diff
@@ -0,0 +1,828 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "9e61ad51-deb2-4e76-9130-8ccf62cf3586",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "for variable in [\"temp\", \"pr\"]:\n",
```
Comment (orianac) [inline notebooks/scrf.ipynb:569]: i feel like all of my comments about temp v. tmax v. tmin might be unnecessary.... but why is it sometimes `temp` and sometimes `tmax`/`tmin`?

Comment (orianac) [reply notebooks/scrf.ipynb:569]: OH i just read the docs at the top again and i understand now. nevermind.

File: notebooks/scrf.ipynb [hunk 4]
```diff
@@ -0,0 +1,828 @@
...
+    "# make sure the different decades are different\n",
+    "# make sure precip is different from tmax, but tmax is the same as tmin\n",
+    "time = 100\n",
+    "\n",
+    "ds1 = xr.open_zarr(\n",
+    "    \"az://flow-outputs/intermediate/scrf/ERA5_pr_2011_2020.zarr\"\n",
+    ").scrf.isel(time=time)\n",
+    "\n",
+    "ds2 = xr.open_zarr(\n",
+    "    \"az://flow-outputs/intermediate/scrf/ERA5_pr_2021_2030.zarr\"\n",
```
Comment (orianac) [inline notebooks/scrf.ipynb:609]: if we wanted to do the full analysis through 2100 we would extend this in decadal increments?

**Review by orianac — COMMENTED** [review]
this was mainly me just learning about this process with small requests for documentation/comments

File: cmip6_downscaling/methods/gard.py (hunk 1 above)
Comment (tcchiao) [reply cmip6_downscaling/methods/gard.py:94]: oh there are def. tweaks

File: notebooks/scrf.ipynb [hunk 5]
```diff
@@ -0,0 +1,828 @@
...
+    "            train_period_end=2020,\n",
+    "            variables=v,\n",
+    "            chunking_approach=None,\n",
+    "        )[v]\n",
+    "\n",
+    "        if v == \"pr\":\n",
+    "            data = data * 1e6\n",
+    "\n",
+    "        # go from 0-360 to -180-180 longitude\n",
+    "        data[\"lon\"] = convert_long3_to_long1(data.lon)\n",
```
[the remark this answers is not in the document]
Comment (tcchiao) [reply notebooks/scrf.ipynb]: probably not needed anymore, will remove

File: notebooks/scrf.ipynb [hunk 6]
```diff
@@ -0,0 +1,828 @@
...
+    "# @dask.delayed\n",
+    "def generate_scrf(\n",
+    "    variable,\n",
+    "    spatial_scales,\n",
+    "    temporal_scales,\n",
+    "    start_year,\n",
+    "    end_year,\n",
+    "):\n",
+    "    if variable == \"temp\":\n",
+    "        path = make_scrf_path(\"ERA5\", \"tasmin\", start_year, end_year)\n",
```
[the remark this answers is not in the document]
Comment (tcchiao) [reply notebooks/scrf.ipynb:435]: this is just for caching so picking one of the two is fine

File: notebooks/scrf.ipynb [hunk 7]
```diff
@@ -0,0 +1,828 @@
...
+    "\n",
+    "    print(\"getting template\")\n",
+    "    template = get_obs(\n",
+    "        obs=\"ERA5\",\n",
+    "        train_period_start=1980,\n",
+    "        train_period_end=1980,\n",
+    "        variables=v,\n",
+    "        chunking_approach=None,\n",
+    "    )[v].isel(time=0)\n",
+    "    template[\"lon\"] = convert_long3_to_long1(template.lon)\n",
```
[the remark this answers is not in the document]
Comment (tcchiao) [reply notebooks/scrf.ipynb]: no, can be deleted

File: notebooks/scrf.ipynb (hunk 4 above)
Comment (tcchiao) [reply notebooks/scrf.ipynb:591]: i generated data from 1850-2100

File: notebooks/scrf.ipynb (hunk 2 above)
Comment (tcchiao) [reply notebooks/scrf.ipynb:475]: added

**Pull request merged**